src/main/java/org/opensearch/security/configuration/SecurityIndexSearcherWrapper.java

```diff
@@ -161,7 +161,7 @@ protected final boolean isBlockedSystemIndexRequest() {
             final TransportAddress caller = threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_REMOTE_ADDRESS);
```

Comment:
wdyt about moving the check to see if the index being acted upon is a system index or not as the first check within the `if (systemIndexPermissionEnabled) { ... }` block? I think the changes in ConfigModel can be avoided if we move the check into here.

Reply:
@cwperks @DarshitChanpura 
Thank you for the comment.
I modified it to perform the check within SecurityIndexSearcherWrapper.java as shown here: [link](https://github.com/opensearch-project/security/pull/4430/files#diff-e4ddeba53461aba177a7c1f823dd1cd9e4137256909b01b55522fbf6a7bf9b75R155). What do you think?